lib/addExcursionPointers.m

```diff
@@ -21,6 +24,8 @@
     %For each excursion set map backtrack through recording each object and
     %the excursions set map it was connected to.
     
+    %The counter variable keeps track of how many key values have currently
+    %been added to the hashmap.
```

Comment:
Thanks for adding this comment in! Could the `counter` variable be removed if we used `{end+1}` each time we need to add a new label/key, e.g.:
```
labelKeys{counter} = node.x_id;
labelValues{counter} = [i];
counter = counter+1;
```
could be replaced by:
```
labelKeys{end+1} = node.x_id;
labelValues{end+1} = [i];
```
?